File: javalin-openapi-processor/src/main/java/net/dzikoysk/openapi/processor/utils/ProcessorUtils.java
package net.dzikoysk.openapi.processor.utils;

import net.dzikoysk.openapi.processor.OpenApiAnnotationProcessor;

import javax.annotation.processing.Messager;
import javax.tools.Diagnostic.Kind;

public final class ProcessorUtils {

    public static void printException(Throwable throwable) {
        Messager messager = OpenApiAnnotationProcessor.getMessager();
        StringBuilder error = new StringBuilder(throwable.getClass() + ": " + throwable.getMessage());

        for (StackTraceElement element : throwable.getStackTrace()) {
            error.append("  ").append(element.toString()).append(System.lineSeparator());
        }

        messager.printMessage(Kind.ERROR, error.toString());

        if (throwable.getCause() != null) {
            messager.printMessage(Kind.ERROR, "---");
            printException(throwable.getCause());
        }
    }

}

File: javalin-openapi-processor/src/main/kotlin/net/dzikoysk/openapi/processor/utils/ProcessorUtils.kt
package net.dzikoysk.openapi.processor.utils

import net.dzikoysk.openapi.processor.OpenApiAnnotationProcessor
import javax.tools.Diagnostic.Kind.ERROR

object ProcessorUtils {

    fun printException(throwable: Throwable) {
        val messager = OpenApiAnnotationProcessor.messager
        val error = StringBuilder(throwable.javaClass.toString() + ": " + throwable.message)

        for (element in throwable.stackTrace) {
            error.append("  ").append(element.toString()).append(System.lineSeparator())
        }

        messager.printMessage(ERROR, error.toString())

        if (throwable.cause != null) {
            messager.printMessage(ERROR, "---")
            printException(throwable.cause!!)
        }
    }

}